data-prepper-plugins/event-json-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/eventjson/EventJsonOutputCodec.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.codec.eventjson;
+
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.codec.OutputCodec;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.sink.OutputCodecContext;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
+import java.util.Objects;
+
+@DataPrepperPlugin(name = "event_json", pluginType = OutputCodec.class, pluginConfigurationType = EventJsonOutputCodecConfig.class)
+public class EventJsonOutputCodec implements OutputCodec {
+    private final ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
+    static final String EVENT_JSON = "event_json";
+    private static final JsonFactory factory = new JsonFactory();
+    private final EventJsonOutputCodecConfig config;
+    private JsonGenerator generator;
+    private OutputCodecContext codecContext;
+
+    @DataPrepperPluginConstructor
+    public EventJsonOutputCodec(final EventJsonOutputCodecConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public String getExtension() {
+        return EVENT_JSON;
+    }
+
+    @Override
+    public void start(OutputStream outputStream, Event event, OutputCodecContext context) throws IOException {
+        Objects.requireNonNull(outputStream);
+        generator = factory.createGenerator(outputStream, JsonEncoding.UTF8);
+        generator.writeStartObject();
```

Comment:
It appears that you are starting an object. But, there is no array. So won't it look like this?

```
{{"data":..., "metadata"...},{"data":..., "metadata"...},{"data":..., "metadata"...}}
```